src/ws/metrics.rs

```diff
@@ -64,24 +148,95 @@ impl Procedure {
     }
 }
 
-pub async fn record_procedure_call(
+/// Records a procedure call. This increments the counter of procedure calls
+/// based on the response code and the specific procedure.
+async fn record_procedure_call(
+    metrics: Arc<Metrics>,
+    code: Option<WsServiceError>,
+    procedure: Procedure,
+) {
+    let code = map_error_code(code);
+    metrics
+        .procedure_call_total_collector
+        .with_label_values(&[code, procedure.as_str()])
+        .inc();
+}
+
+/// Increments the count of connected clients.
+pub async fn increment_connected_clients(metrics: Arc<Metrics>) {
+    metrics.connected_clients_total_collector.inc();
+}
+
+/// Decrements the count of connected clients.
+pub async fn decrement_connected_clients(metrics: Arc<Metrics>) {
+    metrics.connected_clients_total_collector.dec();
+}
+
+/// Records updates sent on subscription. This increments the counter of updates sent
+/// on subscription based on the event type.
+pub async fn record_updates_sent(metrics: Arc<Metrics>, event: FriendshipEvent) {
```

Comment:
can we rename it to be more specific? something that reflects that the updates are on friendship events?